### PR TITLE
Feature/dark mode v3

### PR DESCRIPTION
**Pull request by nhamilton1**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

This change adds dark mode functionality. This will also check the user's system settings and set the theme relative to what their system settings are. The Team and I were highly motivated to find a solution on how to implement this feature, it was mentioned previously that it would be really nice to have so we took action. 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

https://www.loom.com/share/f0d4061a43c841748361fbfdc95f59ba

### CODE REVIEW

File: src/components/common/DarkModeToggle.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    return document.head.querySelector("#antd-stylesheet") || createStylesheetLink();
+};
+
+// checks what the user's systemtheme is, if its dark, it will set the className button toggle so its on.
+const systemTheme = () => {
+    if (window.matchMedia("(prefers-color-scheme: dark)").matches) {
+        document.getElementById("darkModeToggle").className = "ant-switch ant-switch-small ant-switch-checked";
+    };
+    return window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches ? "dark" : "light";
+};
```
Comment (Retrofitt) [inline src/components/common/DarkModeToggle.js:28]: Nice touch checking the user's system theme to automatically set the app theme to dark mode.

File: src/components/common/DarkModeToggle.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+// returns the theme, if no theme, call systemTheme func
+export const getTheme = () => { return localStorage.getItem("theme") || systemTheme(); };
+    export const setTheme = (theme) => {
+        localStorage.setItem("theme", theme);
+        // asssigns the href in getStylesheetLink function to the corrisponding theme in the stylesheets object 
+        getStylesheetLink().href = stylesheets[theme];
+};
```
Comment (Retrofitt) [inline src/components/common/DarkModeToggle.js:36]: It looks like setTheme() is indented to be nested inside getTheme(), but they are both seperate functions to my understanding.

**Review by Retrofitt — APPROVED** [review]
Loom video explains your code well and walks us through your code nicely.

Easy approve for a feature we were all really hard stuck figuring out. Dark mode switch functions well without bugs, code is clean, comments help understand what each function is accomplishing. Overall great work!

**Review by teresafranxman97 — APPROVED** [review]
Love the thorough explanation as well as the comments throughout the code. Readable and easy to understand what every block is doing. Very impressive!

File: src/components/common/DarkModeToggle.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (nhamilton1) [reply src/components/common/DarkModeToggle.js]: Nice catch, fixing that now!

**Review by paulstgermain — COMMENTED** [review]
_Incredible_ work getting a fully functioning Darkmode running here! 🎉

**Pull request merged**